src/app/clusters/general-diagnostics-server/general-diagnostics-server.cpp

```diff
@@ -195,11 +195,24 @@ class GeneralDiagnosticsDelegate : public DeviceLayer::ConnectivityManagerDelega
     }
 
     // Gets called when the device has been rebooted.
-    void OnDeviceRebooted() override
+    void OnDeviceRebooted(uint8_t bootReason) override
```

Comment:
Is there a reason the argument is not BootReasonType?

Reply:
Fair point. Yeah I had some problems with visibility of `BootReasonType` that was declared in the `DiagnosticsDataProvider` class scope, but I moved it to the top of the file and I believe now it should be fine.